controllers/promise_controller.go

```diff
@@ -118,14 +118,14 @@ func (r *PromiseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	}
 	promise := &v1alpha1.Promise{}
 	err := r.Client.Get(ctx, req.NamespacedName, promise)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			r.Log.Info("Promise not found", "namespacedName", req.NamespacedName)
-			return ctrl.Result{}, nil
-		}
+
+	if client.IgnoreNotFound(err) != nil {
 		r.Log.Error(err, "Failed getting Promise", "namespacedName", req.NamespacedName)
 		return defaultRequeue, nil
+	} else if errors.IsNotFound(err) {
+		return ctrl.Result{}, nil
 	}
+
```

Comment:
i like this, but i found the if/else/if a bit confusing :p 

maybe dropping the else?

```go
if errors.IsNotFound(err) {
  return ctrl.Result{}, nil
}
if client.IgnoreNotFound(err) != nil {
  r.Log.Error(err, "Failed getting Promise", "namespacedName", req.NamespacedName)
  return defaultRequeue, nil
}
```

Reply:
Thanks Derik. Will include in the new PR